src/Features/ExternalAccess/OmniSharp/Internal/ExtractClass/OmniSharpExtractClassOptionsService.cs

```diff
@@ -5,38 +5,37 @@
 using System;
 using System.Collections.Immutable;
 using System.Composition;
-using System.Linq;
 using System.Threading;
-using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.ExtractClass;
 using Microsoft.CodeAnalysis.ExtractClass;
+using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Internal.ExtractClass
-{
-    [Shared]
-    [ExportWorkspaceService(typeof(IExtractClassOptionsService))]
-    internal class OmniSharpExtractClassOptionsService : IExtractClassOptionsService
-    {
-        private readonly IOmniSharpExtractClassOptionsService _omniSharpExtractClassOptionsService;
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Internal.ExtractClass;
 
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public OmniSharpExtractClassOptionsService(IOmniSharpExtractClassOptionsService omniSharpExtractClassOptionsService)
-        {
-            _omniSharpExtractClassOptionsService = omniSharpExtractClassOptionsService;
-        }
+[ExportWorkspaceService(typeof(IExtractClassOptionsService)), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class OmniSharpExtractClassOptionsService(
+    IOmniSharpExtractClassOptionsService omniSharpExtractClassOptionsService) : IExtractClassOptionsService
+{
+    private readonly IOmniSharpExtractClassOptionsService _omniSharpExtractClassOptionsService = omniSharpExtractClassOptionsService;
 
-        public async Task<ExtractClassOptions?> GetExtractClassOptionsAsync(Document document, INamedTypeSymbol originalType, ImmutableArray<ISymbol> selectedMembers, CancellationToken cancellationToken)
-        {
-            var result = await _omniSharpExtractClassOptionsService.GetExtractClassOptionsAsync(document, originalType, selectedMembers).ConfigureAwait(false);
-            return result == null
-                ? null
-                : new ExtractClassOptions(
-                    result.FileName,
-                    result.TypeName,
-                    result.SameFile,
-                    result.MemberAnalysisResults.SelectAsArray(m => new ExtractClassMemberAnalysisResult(m.Member, m.MakeAbstract)));
-        }
+    public ExtractClassOptions? GetExtractClassOptions(
+        Document document,
+        INamedTypeSymbol originalType,
+        ImmutableArray<ISymbol> selectedMembers,
+        SyntaxFormattingOptions formattingOptions,
+        CancellationToken cancellationToken)
+    {
+        var result = _omniSharpExtractClassOptionsService.GetExtractClassOptionsAsync(document, originalType, selectedMembers).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
@JoeRobich If you can make the omnisharp ea interface synchronous, this can go away.

Reply:
The implementation is already synchronous in O#. So feel free to update the EA interface and I'll fix things up accordingly with the next insertion.